ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -617,7 +639,8 @@ private BlobReadOptions getDeletedBlobReadOptions(IndexValue value, StoreKey key
       IndexValue putValue =
           findKey(key, new FileSpan(getStartOffset(indexSegments), value.getOffset()), IndexEntryType.PUT,
               indexSegments);
-      if (value.getOriginalMessageOffset() != -1) {
+      if (value.getOriginalMessageOffset() != value.getOffset().getOffset()
```

Comment:
Same as [this](https://github.com/linkedin/ambry/pull/608#discussion_r113814410)